### PR TITLE
actions get actions injected as second arg

### PR DESCRIPTION
**Pull request by lsunsi**
Title is pretty self explanatory.
Change is made according to discussion in issue #107 in the official Mantra Repo.
https://github.com/kadirahq/mantra/issues/107

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -10,7 +10,8 @@ export function injectDeps(context, _actions) {
       const newActionMap = {};
       for (let actionName in actionMap) {
         if (actionMap.hasOwnProperty(actionName)) {
-          newActionMap[actionName] = actionMap[actionName].bind(null, context);
+          const action = actionMap[actionName];
+          newActionMap[actionName] = action.bind(null, context, actions);
```
Comment (arunoda) [inline lib/index.js:14]: It should be `newActionMap`. So, it has the already bounded actions.

**Comment by arunoda** [comment]
I'd like to have some tests as well.

**Comment by chmanie** [comment]
This makes total sense in terms of consistency but introduces a breaking change. Nevertheless I'd love to see this in v2.0! I'd be happy to help out here, write some tests.

**Comment by chmanie** [comment]
Another option would be to put the actions into the context object (which would "just" break for the users that have `actions` already defined in their context for some reason).

**Comment by chmanie** [comment]
Third option: bind the `this` context to all actions. Currently it's the `window` object which doesn't make much sense.

**Comment by chmanie** [comment]
@arunoda I'd like to pursue this further. I'd be happy to provide the implementation as well as tests if you let me know which solution you prefer.

**Comment by chmanie** [comment]
My proposal would be the following:
- For 1.x: bind the `this` context in actions to an object containing context and actions functions so it can be used like this:

``` javascript
export default {
  myAction(context, arg1) {
    const {anotherAction} = this.actions().anotherModule;
    anotherAction('foo'); // already bound to context as well
  }
}
```

For 2.x: Either change the action function bindings so that it can be used like this

``` javascript
myAction(context, actions, arg1, ...) {}
```

or like this (as in composers)

``` javascript
myAction({context, actions}, arg1, ...) {}
```

**Comment by markshust** [comment]
@chmanie what about importing actions as a prop of context:

`myAction({ Meteor, Random, actions }, arg1, ...) {}`

This would retain backwards compatibility, and I think be the most intuitive... no?

**Review by vladejs — APPROVED** [review]

**Pull request closed without merging**